src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs

```diff
@@ -4137,9 +4137,1314 @@ class C1
+                //         x.M<I1>();
+                Diagnostic(ErrorCode.ERR_GenericConstraintNotSatisfiedInterfaceWithStaticAbstractMembers, "M<I1>").WithArguments("C8.M<T8>()", "I1", "T8", "I1").WithLocation(67, 11)
+            };
+
+            compilation2.VerifyDiagnostics(expected);
+
+            compilation2 = CreateCompilation(source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp,
+                                                 references: new[] { compilation1.EmitToImageReference() });
+
+            compilation2.VerifyDiagnostics(expected);
+        }
+
+        [Fact]
+        public void ConstraintChecks_02()
+        {
+            var source1 =
+@"
+public interface I1
+{
+    abstract static void M01();
+}
+
+public class C : I1
+{
+    public static void M01() {}
+}
+
+public struct S : I1
+{
+    public static void M01() {}
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp);
+
+            var source2 =
+@"
+class C1<T1> where T1 : I1
+{
+    void Test(C1<C> x, C1<S> y, C1<T1> z)
+    {
+    }
+}
+
+class C2
+{
+    public void M<T2>(C2 x) where T2 : I1
+    {
+        x.M<T2>(x);
+    }
+
+    void Test(C2 x)
+    {
+        x.M<C>(x);
+        x.M<S>(x);
+    }
+}
+
+class C3<T3> where T3 : I1
+{
+    void Test(C1<T3> z)
+    {
+    }
+}
+
+class C4
+{
+    void M<T4>(C2 x) where T4 : I1
+    {
+        x.M<T4>(x);
+    }
+}
+
+class C5<T5>
+{
+    internal virtual void M<U5>() where U5 : T5 { }
+}
+
+class C6 : C5<I1>
+{
+    internal override void M<U6>() { base.M<U6>(); }
+}
+";
+            var compilation2 = CreateCompilation(source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp,
+                                                 references: new[] { compilation1.ToMetadataReference() });
+
+            compilation2.VerifyEmitDiagnostics();
+
+            compilation2 = CreateCompilation(source2, options: TestOptions.DebugDll,
+                                                 parseOptions: TestOptions.RegularPreview,
+                                                 targetFramework: TargetFramework.NetCoreApp,
+                                                 references: new[] { compilation1.EmitToImageReference() });
+
+            compilation2.VerifyEmitDiagnostics();
+        }
```

Comment:
Consider adding variations of these constraint tests where type parameters are inferred, rather than explicitly specified.

Reply:
> Consider adding variations of these constraint tests where type parameters are inferred, rather than explicitly specified.

I do not see a particular benefit form adding these tests because constraints are going to be checked regardless of how the type argument was provided.

---
In reply to: [606335339](https://github.com/dotnet/roslyn/pull/52366#discussion_r606335339) [](ancestors = 606335339)